### ISSUE TITLE
Freecad hangs when trying to import a step file (which was generated by F360)

### ISSUE THREAD

**Issue by FreeCAD-Bug-Importer** [FreeCAD/FreeCAD#6311]
Issue imported from https://tracker.freecad.org/view.php?id=4832

* **Reporter:** chrisdepalma
* **Date submitted:** 1/16/2022
* **FreeCAD version:** 0.2

# Original report text

I am using latest freecad dev version from 1/16/2022  
Freecad hangs when trying to load the step file.  It does this on the .19.3 version also.  
Freecad starts to render but when clicking on the UI even after an hour after starting import, the ui is unresponsive.  

# Additional information

PC is a I7 5820K with 40 GB of ram.

# Steps to reproduce

* Select file import step.
* Select step file located here:  https://github.com/ZeroGDesign/Mercury/blob/main/CAD/Zero%20G%20-%20Mercury%20One.step

Result: Freecad will never finish import.  

This step was i believe generated by Fusion 360.

# FreeCAD Info

```
OS: Windows 10 Version 2009
Word size of OS: 64-bit
Word size of FreeCAD: 64-bit
Version: 0.19.24267 +99 (Git)
Build type: Release
Branch: Branch_0.19.3
Hash: 6530e364184ce05ccff39501e175cf2237e6ee4b
Python version: 3.8.6+
Qt version: 5.15.2
Coin version: 4.0.1
OCC version: 7.5.3
Locale: English/United States (en_US)

```

## Discussion from Mantis ticket

<details>
  <summary>Click to expand</summary>

### Comment by openBrain 2022-01-17 16:52
chrisdepalma : please report your issues according [our reporting guidelines](https://forum.freecadweb.org/viewtopic.php?f=3&t=5236) by creating a new topic in our forum. Thx

---

### Comment by Kunda1 2022-01-19 16:50
Closing since no forum thread was opened. Re-open if needed. 

---

### Comment by chrisdepalma 2022-01-30 00:12
I added a post on forum as requested.   If anybody cares that step files don't load, then you should try the simple repro steps.

---

### Comment by chrisdepalma 2022-01-30 00:13
post title:  Freecad hangs loading step file.

---

### Comment by Kunda1 2022-01-30 00:51
Y u no add URL ?

---

### Comment by chrisdepalma 2022-01-30 03:49
Kunda1  Url to github is in the issue.

---

</details>

### Forum thread

Comment by adrianinsaval 2022-01-31 20:39

url to the forum thread
here it is: https://forum.freecadweb.org/viewtopic.php?f=3&t=65799
OP please read and follow guidelines when doing a bug report

**Comment by alirezadigi** [comment]
same here
(but not sure of f360)

**Comment by superbonaci** [comment]
CAD Assistant opens it perfectly:

<img width="580" alt="CAD_Assistant" src="https://github.com/FreeCAD/FreeCAD/assets/8339596/b99f1df3-43d3-4bb4-be55-eed68154069f">

**Comment by felipebalbi** [comment]
I have the same issue with 0.21.2. Just tried importing [this](https://openbeagle.org/beagleplay/beagleplay/-/blob/main/Design/beagleplayv10_a2_board_3D_20221219.stp?ref_type=heads) step file and FreeCAD just hangs.

**Comment by superbonaci** [comment]
> I have the same issue with 0.21.2. Just tried importing [this](https://openbeagle.org/beagleplay/beagleplay/-/blob/main/Design/beagleplayv10_a2_board_3D_20221219.stp?ref_type=heads) step file and FreeCAD just hangs.

Same here. The file fully loads and I can do stuff, but after a while of being open the mouse pointer keeps spinning and can't bring the window to front.

If I wait long enough then it un-freezes automatically so kind of works:

![freecad_step](https://github.com/FreeCAD/FreeCAD/assets/8339596/7bf6ca2a-9ef9-40df-b1bd-f291bc32833c)

If you can make another test waiting long enough and check if there's any CPU activity.

**Comment by felipebalbi** [comment]
yeah, I think the step importer is just pretty slow, perhaps, due to the amount of data within the step file? If wait long enough (really long), then it eventually shows the render.

**Comment by superbonaci** [comment]
You can compare the load times with other software like `CAD Assistant`, but remember that FreeCAD is an editor. Software for just viewing usually goes much faster.

**Comment by PaddleStroke** [comment]
Have you guys tried to enable 'skip recompute' before importing the step? Option in right click on the document tree item.

**Comment by superbonaci** [comment]
do not stale.

**Comment by superbonaci** [comment]
do not stale.

**Comment by superbonaci** [comment]
I have some updates now that the bot is complaining. FreeCAD 1.0.2 x86_64 on Windows 11.

@FreeCAD-Bug-Importer file of OP https://github.com/ZeroGDesign/Mercury/blob/main/CAD/Zero%20G%20-%20Mercury%20One.step load fine, but takes 10 to 15 minutes:

<img width="1296" height="920" alt="Image" src="https://github.com/user-attachments/assets/5d355717-3ad7-41ac-abfd-3317a2ddbc3f" />

**Comment by superbonaci** [comment]
@felipebalbi tried today your file https://openbeagle.org/beagleplay/beagleplay/-/blob/main/Design/beagleplayv10_a2_board_3D_20221219.stp?ref_type=heads with FreeCAD 1.0.2 and this is what happens:

It works, but fully loading takes about 30 minutes. The view is loaded earlier but the program is frozen with the mouse pointer circling until fully loads:

<img width="1296" height="920" alt="Image" src="https://github.com/user-attachments/assets/3cf97fec-73d6-4a6d-834a-8e16e4d6feac" />

**Comment by superbonaci** [comment]
I think that what slows down really the import is the fact that it only uses one core of the CPU. I tested the import/open in a computer with Intel 8 cores, and only 1 core hits 100% usage, the others stay just above idling 10% to 40% up and down.

I don't know if it's technically possible to use all the cores during import, but using only one makes it much slower.

**Comment by FEA-eng** [comment]
@superbonaci I guess it's still relevant and should be kept open, right?

**Comment by superbonaci** [comment]
The issue itself is not reproducible anymore, since Freecad does not hang, it's just that takes a long time to load because uses single core. But that's not exactly a bug, it's more a feature request which can introduce new bugs, so not sure if open a new bug for it or now.

This bug can be closed.

**Comment by maxwxyz** [comment]
Closing then.
